sync/_databricks.py

```diff
@@ -885,16 +1026,41 @@ def _get_job_cluster(tasks: List[dict], job_clusters: list) -> Response[dict]:
     return Response(error=DatabricksError(message="Not all tasks use the same cluster"))
 
 
-def _get_run_cluster_id(tasks: List[dict]) -> Response[str]:
-    cluster_ids = {task["cluster_instance"]["cluster_id"] for task in tasks}
-    num_ids = len(cluster_ids)
+def _get_cluster_id_and_tasks_from_run_tasks(
+    all_tasks: List[dict],
+    exclude_tasks: Union[Collection[str], None] = None,
+    project_id: str = None,
+) -> Tuple[str, List[dict]]:
+    cluster_id_to_tasks = {}
+    for task in all_tasks:
+        if not exclude_tasks or task["task_key"] not in exclude_tasks:
+            cluster_id = task["cluster_instance"]["cluster_id"]
+            if cluster_id not in cluster_id_to_tasks:
+                cluster_id_to_tasks[cluster_id] = [task]
+            else:
+                cluster_id_to_tasks[cluster_id].append(task)
+
+    num_ids = len(cluster_id_to_tasks)
+    if num_ids == 0:
+        raise Exception("No cluster found for tasks")
+    elif num_ids > 1:
+        raise Exception("More than 1 cluster found for tasks")
+
+    cluster_id, tasks = cluster_id_to_tasks.popitem()
+    if project_id:
+        project_tasks = [
+            t
+            for t in tasks
+            if t.get("new_cluster", {}).get("custom_tags", {}).get("sync:project-id") == project_id
```

Comment:
Tasks that use a cluster defined in the job won't have a "new_cluster" item. I think we'll need to pull down the cluster for each cluster ID. For example, this run, https://dbc-d95d06ca-1d00.cloud.databricks.com/?o=656201176161048#job/769372317240466/run/14244803, that only used the job cluster:
```
>>> print(orjson.dumps(databricks.get_default_client().get_run(14244803)['tasks'], option=orjson.OPT_INDENT_2).decode())
[
  {
    "run_id": 14245362,
    "task_key": "load_all_lending_tables",
    "run_if": "ALL_SUCCESS",
    "notebook_task": {
      "notebook_path": "/Repos/scott.romney@synccomputing.com/lendingclub/lending_club_medallion_full",
      "source": "WORKSPACE"
    },
    "job_cluster_key": "Job_cluster",
    "state": {
      "life_cycle_state": "TERMINATED",
      "result_state": "SUCCESS",
      "state_message": "",
      "user_cancelled_or_timedout": false
    },
    "run_page_url": "https://dbc-d95d06ca-1d00.cloud.databricks.com/?o=656201176161048#job/769372317240466/run/14245362",
    "start_time": 1693617683584,
    "setup_duration": 312000,
    "execution_duration": 235000,
    "cleanup_duration": 0,
    "end_time": 1693618230789,
    "cluster_instance": {
      "cluster_id": "0902-012124-csylezvn",
      "spark_context_id": "8442236568853351267"
    },
    "attempt_number": 0
  }
]
```